### PR TITLE
Feat/additional assert in unit tests

### LINKED ISSUE #86: Update all unit tests of the parent functions with an extra assert statement
The unit tests for all parent functions listed below should have an extra assert statement to make sure they return the expected output - this is just a final check that the functions are wired correctly:
- `test_get_input_variables` in `tests/utils/test_utils.py`
- `test_get_matrix_a_star` in `tests/matrix_a_star/test_matrix_a_star.py`
- `test_get_match_scores` in `tests/match_scores/test_match_scores.py`
- `test_get_deltas` in `tests/indicator_matrix/test_indicator_matrix.py`
 
This has already been implemented for the `test_get_scaled_labelled_x_star` unit test in `tests/matrix_a_star/test_matrix_a_star.py`, so please look at it for guidance.

### PR DESCRIPTION
**Pull request by grthomson**
## Description
<details><summary>Please include a summary of the changes. </summary>

  - What is this change?
  - Is this a bug fix or a feature and does it break any existing functionality?
  - How has it been tested?
</details>
 
This PR adds some additional assert statements to the following unit tests:

test_get_input_variables in tests/utils/test_utils.py
test_get_matrix_a_star in tests/matrix_a_star/test_matrix_a_star.py
test_get_match_scores in tests/match_scores/test_match_scores.py
test_get_deltas in tests/indicator_matrix/test_indicator_matrix.py

These assert statements check that return values of the parent functions are identical to the returned values of the mocked dependencies. This is just a final check that the functions are wired correctly.

## Type of change

*You can delete options that are not relevant.*

- [ ] Bug fix - *non-breaking change*
- [x] New feature - *non-breaking change*
- [ ] Breaking change - *backwards incompatible change, changes expected behaviour*
- [x] Non-user facing change, structural change, dev functionality, docs ...

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code appropriately, focusing on explaining my design decisions (explain why, not how).
- [x] I have made corresponding changes to the documentation (comments, docstring, etc.. )
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have updated the change log.

<br>

#  Peer review
Any new code includes all the following:

- **Documentation**: docstrings, comments have been added/ updated.
- **Style guidelines**: New code conforms to the project's contribution guidelines.
- **Functionality**: The code works as expected, handles expected edge cases and exceptions are handled appropriately.
- **Complexity**: The code is not overly complex, logic has been split into appropriately sized functions, etc..
- **Test coverage**: Unit tests cover essential functions for a reasonable range of inputs and conditions. Added and existing tests pass on my machine.

### Review comments
Suggestions should be tailored to the code that you are reviewing. Provide context.
Be critical and clear, but not mean. Ask questions and set actions.
<details><summary>These might include:</summary>

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented
  - Do the tests effectively assure that it
  works correctly? Are there additional edge cases/ negative tests to be considered?
- code style improvements (could the code be written more clearly?)
</details>
<br>

*Further reading: [code review best practices](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html)*

### CODE REVIEW

File: tests/utils/test_utils.py [hunk 1]
```diff
@@ -305,6 +305,18 @@ def test_get_input_variables(
     )
     mock_define_kj.assert_called_once_with(cutpoints=test_input_formatted_cutpoints)
     mock_define_K.assert_called_once_with(kj=test_input_kj)
+    assert result["cutpoints"] is mock_format_cutpoints.return_value
```
Comment (danasebo) [inline tests/utils/test_utils.py]: I wonder if it is sufficient to test the return values of these keys (`cutpoints`, `binary_agreement_vars`, ...) from the result dictionary, or all should be tested?

**Review by danasebo — CHANGES_REQUESTED** [review]
Looks good to me, thank you! One small change suggested, but I'll leave up to you if you want to implement it or not.

File: tests/utils/test_utils.py (hunk 1 above)
Comment (grthomson) [reply tests/utils/test_utils.py]: Got it! Have defined the full dictionary as "expected", using return_value of mocked functions as values.

Comment (danasebo) [reply tests/utils/test_utils.py]: thanks Gavin, looks great to me!
The very last thing - and please ignore it if you want as it's just a detail - could we maybe replace `assert * is *` with  `assert * == *` in all added statements (as well as for the `test_get_scaled_labelled_x_star` test added by me) to be consistent with other `assert` statements in the tests of this repo?

Comment (danasebo) [reply tests/utils/test_utils.py]: Agreed with a conversation with Gavin that this is not needed as our understanding is that `assert foo == bar` will check for equality of values, but this check is for identity (in the sense that both sides are literally the same object in memory), thus `assert foo is bar` is more appropriate for those checks of the wiring of functions.

Asked Gavin for one more correction - the rest of the tests use `test_output` for `result` and `expected_output` for `expected`, so please update it for consistency.

File: tests/matrix_a_star/test_matrix_a_star.py [hunk 2]
```diff
@@ -218,7 +219,7 @@ def test_get_scaled_labelled_x_star(
     mock_scale_x_star.return_value = test_scale_x_star_output
 
     # Act
-    expected_output = ma.get_scaled_labelled_x_star(
+    result = ma.get_scaled_labelled_x_star(
```
Comment (danasebo) [inline tests/matrix_a_star/test_matrix_a_star.py]: could this be please changed to `test_output` (also in the assert statement)?

**Review by danasebo — APPROVED** [review]
looks good to me now, many thanks!
